test/functional/rpc_fundrawtransaction.py

```diff
@@ -1200,6 +1201,67 @@ def test_add_inputs_default_value(self):
 
         self.nodes[2].unloadwallet("test_preset_inputs")
 
+    def test_preset_inputs_selection_crash(self):
+        self.log.info('Test wallet preset inputs selection')
+
+        # This exercises the bug presented in #26559 that
+        # ended up on (1) a wallet crash or (2) a created and
+        # broadcasted tx that pays an insane fee amount.
+        #
+        # This is covered by making the wallet selects the preset
+        # inputs twice during the coin selection process (which
+        # is a dangerous bug on v24.0).
+        #
+        # Making the wallet think that the selection process result covers
+        # the entire tx target when it does not. It's actually creating
+        # a tx that sends more coins than what inputs are covering for.
+        #
+        # Which, combined with the subtract fee from outputs option,
+        # makes the wallet generate and broadcast a tx that pays
+        # up to the sum of all the preset inputs, minus one, amounts in
+        # fee :(.
+
+        # Create and fund the wallet with 2 UTXO of 5 BTC each
+        self.nodes[2].createwallet("test_preset_inputs_selection")
+        wallet = self.nodes[2].get_wallet_rpc("test_preset_inputs_selection")
+        outputs = {}
+        for _ in range(2):
+            outputs[wallet.getnewaddress(address_type="bech32")] = 5
+        self.nodes[0].sendmany("", outputs)
+        self.generate(self.nodes[0], 1)
+
+        # Select the preset inputs
+        coins = wallet.listunspent()
+        preset_inputs = [coins[0], coins[1]]
+
+        # Now let's create the tx
+        options = {
+            "inputs": [
+                {
+                    "txid": preset_inputs[0]["txid"],
+                    "vout": preset_inputs[0]["vout"]
+                },
+                {
+                    "txid": preset_inputs[1]["txid"],
+                    "vout": preset_inputs[1]["vout"]
+                },
+            ],
+            "add_inputs": True,
+            "subtract_fee_from_outputs": [0],
+            "add_to_wallet": False
+        }
+
+        # First case, use 'subtract_fee_from_outputs' to make the wallet create a tx that pays an insane fee on v24.0.
+        # (further details in #26559).
+        assert_raises_rpc_error(-4, "Insufficient funds", wallet.send, outputs=[{wallet.getnewaddress(address_type="bech32"): 14}], options=options)
+
+        # Second case, don't use 'subtract_fee_from_outputs' to make the wallet crash due the insane fee on v24.0.
+        # (further details in #26559).
```

Comment:
```suggestion
        # Attempt to send 14 BTC. The wallet should exclude the preset inputs from the pool of
        # available coins, realize that there is not enough money to fund the 14 BTC payment,
        # and fail with "insufficient funds."
        # Even with SFFO, the wallet can only afford to send 10 BTC.

        # If the wallet does not properly exclude preset inputs from the pool of available coins
        # prior to coin selection, it may create a transaction that does not fund the full payment
        # amount or, through SFFO, incorrectly reduce the payment amount so that payment+fee is not
        # equal to 14 BTC.
        assert_raises_rpc_error(-4, "Insufficient funds", wallet.send, outputs=[{wallet.getnewaddress(address_type="bech32"): 14}], options=options)
```

Reply:
Updated, thanks